lib/mayaUsd/fileio/translators/translatorMaterial.cpp

```diff
@@ -51,6 +58,48 @@
 
 PXR_NAMESPACE_OPEN_SCOPE
 
+TF_DEFINE_PRIVATE_TOKENS(_tokens, (inputs)(varname));
+
+namespace {
+// We want to know if this material is a specialization that was created to handle UV mappings on
+// export. For details, see the _UVMappingManager class in ..\shading\shadingModeExporterContext.cpp
+//
+// If the root layer contains anything that is not a varname input, we do not consider it a
+// mergeable material.
+bool _IsMergeableMaterial(const UsdShadeMaterial& shadeMaterial)
+{
+    if (!shadeMaterial || !shadeMaterial.HasBaseMaterial()) {
+        return false;
+    }
+    const PcpPrimIndex&   primIndex = shadeMaterial.GetPrim().GetPrimIndex();
+    PcpNodeRef            primRoot = primIndex.GetRootNode();
+    SdfPath               primPath = primIndex.GetPath();
+    const SdfLayerHandle& layer = primRoot.GetLayerStack()->GetLayerTree()->GetLayer();
```

Comment:
I'm not sure I understand why we're trying to find the root layer here. Don't we want to be considering the fully composed stage? Materials could be added or specialized, or opinions could be expressed, across multiple layers.